torch/csrc/autograd/python_function.cpp

```diff
@@ -865,12 +871,23 @@ std::pair<UnpackedInput, InputFlags> unpack_input(PyObject* args) {
       }
       Py_INCREF(Py_False);
       PyTuple_SET_ITEM(flags.needs_input_grad.get(), i, Py_False);
+
+      if (need_input) {
+        // The following conversion from PyObject to IValue is expensive
+        // Only do it if profiler is enabled and needs input shapes
+        auto match = torch::jit::tryToInferType(arg);
+        if (match.success()) {
```

Comment:
If the idea here to want to record ALL arguments so you can replay it later, what happens later when there's a type you cannot infer?

Reply:
The type that can not be inferred are nn.Module and ScriptModule:
1. https://www.internalfb.com/code/fbsource/[ef3e69d17cea50617109f1c9b47d29db0891b443]/fbcode/caffe2/torch/csrc/jit/python/pybind_utils.h?lines=489
2. https://www.internalfb.com/code/fbsource/[ef3e69d17cea50617109f1c9b47d29db0891b443]/fbcode/caffe2/torch/csrc/jit/python/pybind_utils.h?lines=495

I do not see a use case that an operator need these types to run. The use case I want to support is having some primitive data types or their container to define an op. For example, padding and strides in a convolution op.

Reply:
Do you have a code pointer to how this is normally done? I think we want to make sure they do the same thing.